### PR TITLE
Add docs on API Development Strategy

### PR DESCRIPTION
**Pull request by brophdawg11**
Initial pass at future flags docs

[Rendered](https://github.com/remix-run/remix/blob/brophdawg11/future-flags-docs/docs/pages/api-development-strategy.md)

### CODE REVIEW

File: docs/pages/api-development-strategy.md [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+
+**`future.v2_feature`**
+
+`v2_` indicates a breaking change from v1 behavior and implies (1) that the API is considered stable and will not under any more breaking changes and (2) that the API will become the default behavior in v2. A `v2_` flag _does not_ mean the feature is bug-free - no software is! Our recommendation is to upgrade to v2 flags as you have the time, as it will make your v2 upgrade _much_ smoother.
+
+### Example New Feature Flow
+
+The decision flow for a new feature looks something like this:
+
+![Flowchart of the decision process for how to introduce a new feature][feature-flowchart]
```
Comment (brophdawg11) [inline docs/pages/api-development-strategy.md:52]: Image lives over in the website repo: https://github.com/remix-run/remix-website/blob/brophdawg11/future-flags/public/docs-images/feature-flowchart.png

File: docs/pages/api-development-strategy.md [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+
+The lifecycle is thus either:
+
+- Non-Breaking + Stable API Feature -> Lands in v1
+- Non-Breaking + Unstable API -> `future_unstable_` flag -> Lands in v1
```
Comment (MichaelDeBoey) [inline docs/pages/api-development-strategy.md:57]:
```suggestion
- Non-Breaking + Unstable API -> `future.unstable_` flag -> Lands in v1
```

File: docs/pages/api-development-strategy.md [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+- Breaking + Stable API Feature -> `future.v2_` flag -> Lands in v2
+- Breaking + Unstable API -> `future_unstable_` flag -> `future.v2_` flag -> Lands in v2
```
Comment (MichaelDeBoey) [inline docs/pages/api-development-strategy.md:59]:
```suggestion
- Breaking + Unstable API -> `future.unstable_` flag -> `future.v2_` flag -> Lands in v2
```

File: docs/pages/api-development-strategy.md [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+| `v2_errorBoundary`              | Combine `ErrorBoundary`/`CatchBoundary` into a single `ErrorBoundary` |
+| `v2_meta`                       | Enable the new API for your `meta` functions                          |
+| `v2_routeConvention`            | Enable the flat routes style of file-based routing                    |
+
+We're in the process of preparing for our v2 release, so all `future.unstable_` flags are being stabilized into `future.v2_` flags (except for those which are not breaking changes, like PostCSS/Tailwind/Vanilla Extract support). This includes adding deprecation warnings for apps still using the old way. Once we stabilize them all we'll do a final Remix 1.15.0 release and let that run for a bit to give folks time to opt into any flags they haven't yet added. Then we'll plan to release Remix 2.0.0 and start working on releasing flag-driven Remix v3 features.
+
+[future-flags-blog-post]: https://remix.run/blog/future-flags
+[react-router-v6-migration-guide]: https://reactrouter.com/en/main/upgrading/v5#introduction
+[react-router-v6-back-compat]: https://reactrouter.com/en/main/upgrading/v5#backwards-compatibility-package
```
Comment (MichaelDeBoey) [inline docs/pages/api-development-strategy.md:81]:
```suggestion
[react-router-v6-migration-guide]: https://reactrouter.com/upgrading/v5#introduction
[react-router-v6-back-compat]: https://reactrouter.com/upgrading/v5#backwards-compatibility-package
```

**Review by MichaelDeBoey — CHANGES_REQUESTED** [review]

**Review by MichaelDeBoey — APPROVED** [review]

**Pull request merged**